src/BenchmarkDotNet/Engines/StoppingCriteria/AutoWarmupStoppingCriteria.cs

```diff
@@ -0,0 +1,97 @@
+        /// </summary>
+        /// <param name="minIterationCount">
+        /// The minimum number of iterations.
+        /// We are always going to do at least <paramref name="minIterationCount"/> iterations regardless of fluctuations.
+        /// </param>
+        /// <param name="maxIterationCount">
+        /// The maximum number of iterations.
+        /// We are always going to do at most <paramref name="maxIterationCount"/> iterations regardless of fluctuations.
+        /// </param>
+        /// <param name="minFluctuationCount">
+        /// The required number of fluctuations.
+        /// If the required number of fluctuations is achieved but the number of iterations less than <paramref name="minIterationCount"/>,
+        /// we need more iterations.
+        /// If the required number of fluctuations is not achieved but the number of iterations equal to <paramref name="maxIterationCount"/>,
+        /// we should stop the iterations.
+        /// </param>
+        public AutoWarmupStoppingCriteria(int minIterationCount, int maxIterationCount, int minFluctuationCount = DefaultMinFluctuationCount)
+        {
+            this.minIterationCount = minIterationCount;
+            this.maxIterationCount = maxIterationCount;
+            this.minFluctuationCount = minFluctuationCount;
+
+            maxIterationMessage = $"The maximum amount of iteration ({maxIterationCount}) is achieved";
+            minFluctuationMessage = $"The minimum amount of fluctuation ({minFluctuationCount}) and " +
+                                    $"the minimum amount of iterations ({minIterationCount}) are achieved";
+        }
+
+        protected override string GetTitle() => $"{nameof(AutoWarmupStoppingCriteria)}(" +
+                                                $"{nameof(minIterationCount)}={minIterationCount}, " +
+                                                $"{nameof(maxIterationCount)}={maxIterationCount}, " +
+                                                $"{nameof(minFluctuationCount)}={minFluctuationCount})";
+
+        protected override int GetMaxIterationCount() => maxIterationCount;
+
+        protected override IEnumerable<string> GetWarnings()
+        {
+            if (minIterationCount < 0)
+                yield return $"Min Iteration Count ({minIterationCount}) is negative";
+            if (maxIterationCount < 0)
+                yield return $"Max Iteration Count ({maxIterationCount}) is negative";
+            if (minFluctuationCount < 0)
+                yield return $"Min Fluctuation Count ({minFluctuationCount}) is negative";
+            if (minIterationCount > maxIterationCount)
+                yield return $"Min Iteration Count ({minFluctuationCount}) is greater than Max Iteration Count ({maxIterationCount})";
+        }
+
+        public override StoppingResolution Evaluate(IReadOnlyList<Measurement> measurements)
```

Comment:
nit: I would put this public method to the top of the file (right after the ctor), this is the most important piece of code here

Reply:
Good idea, will do it.